File: Clase 18/PYTHON/ejemplo_animales/Perro.py
from Animal import Animal

class Perro(Animal):

    def __init__(self, id, nombre, especie, poder_mandibula) -> None:
        super().__init__(id, nombre, especie)
        self.poder_mandibula = poder_mandibula
    
    def hace_guau(self):
        print(f"{self.nombre} hace GUAUUUU!!!!")

    # Polimorfismo por Sobrecarga
    def se_lava(self):
        print(f"{self.nombre} se lava en una tina...")
    
    # Polimorfismo por inclusion
    def hace_sonido(self):
        self.hace_guau()

File: Clase 18/JAVA/ejemploAnimales/Perro.java
package ejemploAnimales;

public class Perro extends Animal{
    protected int poderMandibula;

    public Perro(int id, String nombre, String especie, int poderMandibula){
        super(id, nombre, especie);
        this.poderMandibula = poderMandibula;
    }

    public void haceGuau(){
        System.out.println(this.nombre + " hace GUAU!!!!");
    }

    // Polimorfismo por sobrecarga
    public void seLava(){
        System.out.println(this.nombre + " se lava en una tina...");
    }

    // Polimorfismo Por Inclusion
    @Override
    public void haceSonido(){
        this.haceGuau();
    }
}